src/Sentry/Internal/OriginHelper.cs

```diff
@@ -0,0 +1,37 @@
+namespace Sentry.Internal;
+
+internal static partial class OriginHelper
+{
+    internal const string Manual = "manual";
+    private const string ValidOriginPattern = @"^(auto|manual)(\.[\w_-]+){0,3}$";
```

Comment:
```suggestion
    private const string ValidOriginPattern = @"^(auto|manual)(\.[\w]+){0,3}$";
```
The hyphen is not allowed. And the underscore is already part of `\w`.